### DISCUSSION TITLE
Problem with compiling your application

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#16630] [Q&A Getting Started] [answered]
Hi everyone. I am a new learner for Moose. When I try to compile and test your application on this page (https://mooseframework.inl.gov/getting_started/new_users.html#new-users) I got the following issue.
![image](https://user-images.githubusercontent.com/74226455/103921564-48d4dd80-50d8-11eb-8994-23bc267a139c.png)

I do not know how to solve it.

**Comment by cticenhour** [comment, marked as answer]
You need to first create an application in your `projects` directory. Then you can build and test it. The intent of the directions was for you to create an app with a name you decide (replace `YourAppName` with another name, which could be anything you want). 

Assuming that you have installed and tested MOOSE itself successfully, you need to follow the directions on that page closely. I'll recap here. Each line should be entered into your prompt separately. 

```
cd projects
./moose/scripts/stork.sh YourAppName  # REPLACE YourAppName WITH A CUSTOM NAME
```

Then you can perform each of the three commands you have in your original question, replacing `YourAppName` with the name of your app.

**Reply by aeslaughter** [reply]
You can also take a look at our tutorial: https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/index.html

**Reply by xchengood** [reply]
> You need to first create an application in your `projects` directory. Then you can build and test it. The intent of the directions was for you to create an app with a name you decide (replace `YourAppName` with another name, which could be anything you want).
> 
> Assuming that you have installed and tested MOOSE itself successfully, you need to follow the directions on that page closely. I'll recap here. Each line should be entered into your prompt separately.
> 
> ```
> cd projects
> ./moose/scripts/stork.sh YourAppName  # REPLACE YourAppName WITH A CUSTOM NAME
> ```
> 
> Then you can perform each of the three commands you have in your original question, replacing `YourAppName` with the name of your app.

Thank you. This issue does not appear according to your instructions. But I got another error.

![image](https://user-images.githubusercontent.com/74226455/104047035-99b30780-51a6-11eb-997a-018241878d01.png)

Do I need to install 'libmesh-config'? Could you tell me how to install it?

**Reply by xchengood** [reply]
> You can also take a look at our tutorial: https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/index.html

Thank you. I am studying this step by step.

**Reply by milljm** [reply]
you are operating in the (base) environment as denoted in your screen shot. You'll want to first activate the moose environment and then run make:

```bash
 (base) xinchen@xins-mbp apple % conda activate moose
 (moose) xinchen@xins-mbp apple %
# ^^^^^ moose environment (and thus libmesh) is available. Now you can run make:

cd ~/projects/moose/test
make -j4
```
If make succeeds, then you can `./run_tests -j 4`

**Reply by cticenhour** [reply]
> Thank you. This issue does not appear according to your instructions. But I got another error.

This means that it cannot find libMesh. You are running in the "base" conda environment, so you must activate your moose environment whenever you want to do moose-based development. Run `conda activate moose` in your Terminal and then try your build and test commands again.

**Reply by xchengood** [reply]
> you are operating in the (base) environment as denoted in your screen shot. You'll want to first activate the moose environment and then run make:
> 
> ```shell
>  (base) xinchen@xins-mbp apple % conda activate moose
>  (moose) xinchen@xins-mbp apple %
> # ^^^^^ moose environment (and thus libmesh) is available. Now you can run make:
> 
> cd ~/projects/moose/test
> make -j4
> ```
> 
> If make succeeds, then you can `./run_tests -j 4`

Thank you. I fix this issue. When I try Mesh in 'Example 1 : As Simple as it Gets',
![image](https://user-images.githubusercontent.com/74226455/104490247-39e9a180-5596-11eb-812a-4c60cd8858f8.png)

I got the following issue. Could you help me solve it? Thank you.

![image](https://user-images.githubusercontent.com/74226455/104490341-5c7bba80-5596-11eb-86b6-4960c725eda6.png)

**Reply by aeslaughter** [reply]
This section just explaining what is in the input file, go down to the "Running the Problem" section to execute the example.

**Reply by aeslaughter** [reply]
You should also consider going through our tutorial carefully, it will help you understand how MOOSE is intended to work.
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/index.html
